hidden/Features/Preferences/PreferencesViewController.swift

```diff
@@ -38,8 +38,8 @@ class PreferencesViewController: NSViewController {
     }
     
     //MARK: - VC Life cycle
-    override func viewDidLoad() {
-        super.viewDidLoad()
+    override func viewDidAppear() {
```

Comment:
Why would you need to change to viewDidAppear 🤔

Reply:
You can change it back to `viewDidLoad()` and do the following things:
1. turn off auto hide in the preferences
2. close the preferences window and enable auto hide in the status menu
3. go to preferences again

The auto hide checkbox will not be updated.

Reply:
`viewDidLoad` only executes when the window is loaded. `viewDidAppear` executes every time when the window is activated.

In the case of hidden bar, the prefs window only hides instead of closes. Which means `viewDidLoad` will only be executed once, when the window is initialized. But you need to make sure that the auto hide checkbox will update every time when the window appears.

Reply:
Ok I see, but it would not solve the problem sync state between the preference ui and menu bar (see my attached image). I think you should use The KVO instead 🤔 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/14150687/85815355-6b358800-b792-11ea-94e4-d6a397656a55.png">

Reply:
I'm not sure how to use KVO to observe a static property. As far as I know, the observed property has to be marked as dynamic.

Reply:
@p-z-l you can use NotificationCenter to observe when the setting changed in Preferences 👍

Reply:
Problem solved: <a href="https://github.com/dwarvesf/hidden/pull/87/commits/eb68047ef02db01e88f7d24cf5d07be9e804cc5a">eb68047</a>